frontend/lib/search/filter_bar_button.dart

```diff
@@ -56,14 +60,30 @@ class _FilterBarButtonState extends State<FilterBarButton>
 
   @override
   Widget build(BuildContext context) {
+    const maxButtonWidth = 80.0;
     const paddingPerElement = 8;
     const minNumberElements = 5;
+
+    // At least a width of 65 is needed to fit all category names
+    const requiredCategoryLongTitleWidth = 65.0;
+
     var totalWidth = MediaQuery.of(context).size.width;
-    var width = min(
-        80.0,
+
+    var smallWidth = min(
+        maxButtonWidth,
+        // Width allowing at least minNumberElements in one row
         (totalWidth - minNumberElements * paddingPerElement) /
             minNumberElements);
 
+    var largeWidth = max(requiredCategoryLongTitleWidth, smallWidth);
+
+    var numLargeElementsPerRow = totalWidth ~/ largeWidth;
+    var isSecondRow =
+        widget.index >= max(minNumberElements, numLargeElementsPerRow);
```

Comment:
If I understand correctly, `max(minNumberElements, numLargeElementsPerRow)` is the number of elements in the first row. But why do we use `numLargeElementsPerRow` here as in the first row only smallWidth is used?

Shouldn't it rather be `max(minNumberElements, numSmallElementsPerRow)` with `numSmallElementsPerRow = totalWidth ~/ smallWidth`?